config/application.rb

```diff
@@ -50,5 +50,8 @@ class Application < Rails::Application
 
     # Defer loading of images until it reaches a calculated distance from the viewport
     config.action_view.image_loading = "lazy"
+
+    config.view_component.default_preview_layout = "components_previews"
+    config.view_component.preview_paths << "#{Rails.root}/app/components/previews"
```

Comment:
Here are the two main configurations that I needed to change.

Line 54 is the most important one, where I'm configuring to use the new layout that I added. We needed that for two reasons, first because if we use the main layout the component will be rendered using the header and footer of the main site and probably we don't want that and secondly because we use helpers defined in the ApplicationController but that controller is not used here, so they're not accessible. Other solution for that is to use a custom controller but I decided by this solution because either way we needed a new layout.

Line 55 is simply to store previews on `app/components/previews` instead of `test/components/previews`. I think this option is better because it keeps the previews relatively near to their definition.

Reply:
Let's use `component_preview` as the name of the component layout like [the docs recommend](https://lookbook.build/guide/previews/layouts/).

I'd rather keep the code under `test/` but if we ever make this available in production then `app/` might make more sense. Either way, the docs recommend `test/` so let's keep it there for now.

Reply:
For what it is worth, my prefered approach it to use sidecar component organisation and put the previews right next to the component. 
- previews are not really related to test 
- Previews are helpful when developpping a component (to fine tune the look and feel) so having them close to the dev file can make sens
- it is easier when you need to display the design system in staging or production as usually the test forlder is ignored in those envs

Some more info on that https://github.com/palkan/view_component-contrib#organizing-components-or-sidecar-pattern-extended